src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTaskTest.java

```diff
@@ -89,25 +92,37 @@ private UploadEnvelopeDocumentsTask getTask(long minimumEnvelopeAge) {
         );
     }
 
-    private Envelope getEnvelope(String containerName) {
-        return new Envelope(
-            "po-box",
-            "jurisdiction",
-            now(), // delivery date
-            now(), // opening date
-            now(), // zip file created at (from blob storage)
-            ZIP_FILE_NAME,
-            "case-number",
-            "previous-service-case-reference",
-            Classification.EXCEPTION,
-            emptyList(),
-            emptyList(),
-            emptyList(),
-            containerName
-        );
+    private Envelope getEnvelope(String containerName, long amountOfMinutesToSubtract) {
+        try {
+            Envelope envelope = new Envelope(
+                "po-box",
+                "jurisdiction",
+                now(), // delivery date
+                now(), // opening date
+                now(), // zip file created at (from blob storage)
+                ZIP_FILE_NAME,
+                "case-number",
+                "previous-service-case-reference",
+                Classification.EXCEPTION,
+                emptyList(),
+                emptyList(),
+                emptyList(),
+                containerName
+            );
+
+            Field field = Envelope.class.getDeclaredField("createdAt");
+            field.setAccessible(true);
+            field.set(envelope, now().minus(amountOfMinutesToSubtract, MINUTES));
+
+            return envelope;
+        } catch (NoSuchFieldException | SecurityException exception) {
+            throw new RuntimeException("Could not access 'createdAt' field in envelope", exception);
+        } catch (IllegalArgumentException | IllegalAccessException exception) {
+            throw new RuntimeException("Could not update 'createdAt' field in envelope", exception);
+        }
```

Comment:
This can probably be just one `catch(Exception exception)`

Reply:
I'm being a good boy here and explicitly defining whats wrong with what where

Reply:
indeed or can throws Exception as it is private method and just used above :)

Reply:
Helped me out actually as I forgot set access and got specific message. Plus - I'll remove those pretty soon anyway